doc/source/how-to-use-driver-api.rst

```diff
@@ -0,0 +1,254 @@
+            log(INFO,f"Got {len(all_node_ids)} client nodes: {all_node_ids}")
+            if len(all_node_ids) >= num_client_nodes_per_round:
+                # Sample client nodes
+                sampled_nodes = random.sample(all_node_ids, num_client_nodes_per_round)
+                break
+            time.sleep(3)
+
+        # Log sampled node IDs
+        log(INFO,f"Sampled {len(sampled_nodes)} node IDs: {sampled_nodes}")
+
+        # Do something else
+    ...
+
+
+Push messages to nodes
+----------------------
+
+Now that we have a list of node IDs, we can push information to them to execute a task such as training or evaluating a model. To achieve this, we will first use :code:`Driver.create_message()` to create a :code:`Message` and then use :code:`Driver.push_message()` to push the :code:`Messages` to the list of node IDs. 
+
+.. admonition:: Note
+    :class: note
+
+    Each :code:`Message` contains a |recordset_link|_ object. It contains :code:`parameters_records`, :code:`metrics_records`, and :code:`configs_records` attributes, which are - unsurprisingly - parameters, metrics, and configurations that are used by a node to execute a task. 
+
+Here is an example of how to push a PyTorch model and instructions to a set of client node IDs to train the model. First, we define a utility function to convert a PyTorch model into a :code:`ParametersRecord` and create a PyTorch model in :code:`main()` (we’ve omitted the implementation details for :code:`Net()`, but you can refer to :doc:`this quickstart tutorial <tutorial-quickstart-pytorch>` for an example):
+
+.. code-block:: python
+
+    # File: server.py
+
+    def pytorch_to_parameter_record(pytorch_module: torch.nn.Module):
+        state_dict = pytorch_module.state_dict()
+        for k, v in state_dict.items():
+            state_dict[k] = _ndarray_to_array(v.numpy())
+        return ParametersRecord(state_dict)
+
+
+    @app.main()
+    def main(driver: Driver, context: Context) -> None:
+    	global_model = Net()
+
+Then, we create a :code:`RecordSet` and add parameters and configurations to :code:`parameters_record` and :code:`configs_record`, respectively. Note that the dictionary keys used here are customizable, so you have a great degree of flexibility to assign and use the dictionaries in a :code:`RecordSet`:
+
+.. code-block:: python
+
+    # File: server.py
+    # In the for-loop in `main()`
+
+    # Create a RecordSet
+    recordset = RecordSet()
+
+    # Add model parameters to the RecordSet
+    recordset.parameters_records["my_model"] = pytorch_to_parameter_record(global_model)
+
+    # Add a training configuration for 1 epoch only to the RecordSet
+    recordset.configs_records["my_config"] = ConfigsRecord({"epochs": 1})
+
+Next, we create a list of :code:`Messages`, one for each node ID. To do so, we loop over all node IDs and run :code:`Driver.create_message()` with the :code:`recordset` as the content of the message: 
```

Comment:
```suggestion
Next, we create a list of :code:`Message` objects, one for each node ID. To do so, we loop over all node IDs and run :code:`Driver.create_message()` with the :code:`recordset` as the content of the message: 
```

Reply:
I'm not sure what's better to do:
- `Message` is the class name. If we want to use plural, "`Message`s" looks terrible. I think "`Message` objects" is a better way.
- Saying "`Messages`" I think is not ideal either because there is not a class named `Messages`.

What do you think? What's the preferred approach?

Reply:
@jafermarq 💯 I prefer the first approach. Will update throughout.